cvat/apps/engine/backup.py

```diff
@@ -114,26 +114,21 @@ def _read_annotation_guide(zip_object, guide_filename, assets_dirname):
 
     return None, []
 
-def _import_annotation_guide(guide_data, assets):
+def _import_annotation_guide(owner, guide_data, assets):
     guide_serializer = AnnotationGuideWriteSerializer(data=guide_data)
     markdown = guide_data['markdown']
     if guide_serializer.is_valid(raise_exception=True):
         guide_serializer.save()
 
     for asset in assets:
         name, data = asset
-        basename = os.path.basename(name)
-        asset_serializer = AssetWriteSerializer(data={
-            'filename': basename,
-            'guide_id': guide_serializer.instance.id,
-        })
-        if asset_serializer.is_valid(raise_exception=True):
-            asset_serializer.save()
-            markdown = markdown.replace(f'{name}', f'/api/assets/{asset_serializer.instance.pk}')
-            path = os.path.join(settings.ASSETS_ROOT, str(asset_serializer.instance.uuid))
-            os.makedirs(path)
-            with open(os.path.join(path, basename), 'wb') as destination:
-                destination.write(data)
+        asset_serializer = AssetWriteSerializer.write_asset(
+            owner,
+            guide_serializer.instance.id,
+            data,
+            os.path.basename(name)
```

Comment:
I would suggest using `SimpleUploadedFile` here. That way, `AssetWriteSerializer` will only need to handle one input data type. In addition, `SimpleUploadFile` will automatically sanitize the name.